spec/index.bs

```diff
@@ -12090,19 +12090,63 @@ of WebGPU calls, typically for debugging purposes or to make an operation more f
 
 <script type=idl>
 enum GPUErrorFilter {
-    "out-of-memory",
-    "validation"
+    "validation",
+    "internal"
 };
 </script>
 
+{{GPUErrorFilter}} defines the type of errors that should be caught when calilng
+{{GPUDevice/pushErrorScope()}}:
+
+<dl dfn-type=enum-value dfn-for=GPUPrimitiveTopology>
+    : <dfn>"validation"</dfn>
+    ::
+        Indicates that an operation did not satisfy all validation requirements. Validation errors
+        are always indicitive of an application error, and is expected to fail the same way across
+        all devices assuming the same {{device/[[features]]}} and {{device/[[limits]]}} are in use.
+
+    : <dfn>"internal"</dfn>
+    ::
+        Indicates than an operation failed for a system or implementation-specific reason even
+        when all validation requirements have been satisfied. The same operation may succeed on
+        other devices or under difference circumstances.
+
+        This error may indicate there was not enough free memory to complete the requested operation
+        or may be due to the operation exceeding the capabilities of the implementation in a way not
+        easily captured by the [=supported limits=].
+
+        If the failed operation was a resources allocation, such as the creation of a texture or
+        buffer, it may succeed if attempted again with a lower memory requirement (like using
+        smaller texture dimensions), or if memory used by other resources is released first.
+        Otherwise the operation may succeed if attempted again with changes to lower the complexity,
+        such as using a simpler shader during pipeline creation.
+</dl>
+
+<div class=note>
+Note: Error scopes can encompass as many commands as needed. The number of commands an error scope covers
+will generally be correlated to what sort of action the application intends to take in response to
+an error occuring.
+
+For example: An error scope that only contains the creation of a single resource, such as a texture
+or buffer, can be used to detect failures such as out of memory conditions, in which case the
+application may try freeing some resources and trying the allocation again.
+
+Error scopes do not indetify which command failed, however. So, for instance, wrapping all the
+commands executed while loading a model in a single error scope will not offer enough granularity to
+determine if the issue was due to memory constraints. As a result freeing resources would usually
+not be a productive response to a failure of that scope. A more appropriate response would be to
+allow the application to fall back to a different model or produce a warning that the model could
+not be loaded.
```

Comment:
Perhaps suggest here that a smaller nested error scope could be used for this.

Reply:
Done!